private/rules/v1_lock_file.bzl

```diff
@@ -0,0 +1,166 @@
+def _remove_auth_from_url(url):
+    """Returns url without `user:pass@` or `user@`."""
+    if "@" not in url:
+        return url
+    protocol, url_parts = _split_url(url)
+    host = url_parts[0]
+    if "@" not in host:
+        return url
+    last_index = host.rfind("@", 0, None)
+    userless_host = host[last_index + 1:]
+    new_url = "{}://{}".format(protocol, "/".join([userless_host] + url_parts[1:]))
+    return new_url
+
+def _is_valid_lock_file(lock_file_contents):
```

Comment:
Should we also apply a recursive spec check on the expected keys in the dependency_tree object?

Reply:
Agreed. This does need more checks in it :) Done!